nomad/acl_endpoint.go

```diff
@@ -1724,6 +1724,20 @@ func (a *ACL) UpsertAuthMethods(
 		return err
 	}
 
+	// Populate the response. We do a lookup against the state to pick up the
+	// proper create / modify times.
+	stateSnapshot, err := a.srv.State().Snapshot()
+	if err != nil {
+		return err
+	}
+	for _, method := range args.AuthMethods {
+		lookupAuthMethod, err := stateSnapshot.GetACLAuthMethodByName(nil, method.Name)
+		if err != nil {
+			return structs.NewErrRPCCodedf(400, "ACL auth method lookup failed: %v", err)
+		}
+		reply.AuthMethods = append(reply.AuthMethods, lookupAuthMethod)
```

Comment:
`GetACLAuthMethodByName` can return nil without an error if the method doesn't exist. Practically speaking I can't see this happening but if we had a bug that allowed `raftApply` to return without error but not write the method, we could be appending a nil pointer to the results (which would crash in the CLI because the formatted doesn't consider the nil case).

For belt-and-suspenders can we check that `lookupAuthMethod` isn't nil before adding it to the reply?